### ISSUE TITLE
High Level Support of Apex SOAP API?

### ISSUE THREAD

**Issue by Krisa** [jsforce/jsforce#565]
It would be great to have, at least, an high level support of the Apex SOAP API (WSDL to be found here /services/wsdl/apex)

That's at the end very similar to the Enterprise/Partner API but the endpoint is `/services/Soap/s/` instead of `/services/Soap/u/` for the Partner API.

**Comment by stomita** [comment]
JSforce does support some of SOAP partner api methods. See the [doc](http://jsforce.github.io/jsforce/doc/SoapApi.html).
If you prefer the support of WSDL, JSforce is not selecting the approach because the WSDL spec is complicated and relatively heavy just for doing Salesforce API call.

**Comment by Krisa** [comment]
Right, but the request is just to have a high level support of it. At least the endpoint :). And for the rest, we could just build our own request and call an "invoke" method, but the purpose would be to leverage useful built-in jsforce mechanism, such as most essentially the auto-refreshing of session with access/refresh token.

**Comment by Krisa** [comment]
Just re-reading your comment. I'm referring to the Apex API, which is not related to either the Partner or Enterprise SOAP:
![image](https://cloud.githubusercontent.com/assets/1253866/20575170/6f21638c-b1b8-11e6-9136-1155653bb542.png)

**Comment by stomita** [comment]
OK understand the api for managing Apex classes and triggers.

**Comment by Krisa** [comment]
Indeed. Can you please review the PR as it implements the first method of the Apex SOAP API ? @stomita
